src/EditorFeatures/CSharpTest/ChangeSignature/AddParameterTests.Formatting.cs

```diff
@@ -0,0 +1,490 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ChangeSignature;
+using Microsoft.CodeAnalysis.Editor.UnitTests.ChangeSignature;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities.ChangeSignature;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ChangeSignature
+{
+    public partial class ChangeSignatureTests : AbstractChangeSignatureTests
+    {
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ChangeSignature)]
+        public async Task AddParameter_Formatting_KeepCountsPerLine()
+        {
+            var markup = @"
+class C
+{
+    void $$Method(int a, int b, int c,
+        int d, int e,
+        int f)
+    {
+        Method(1,
+            2, 3,
+            4, 5, 6);
+    }
+}";
+            var updatedSignature = new[] {
+                new AddedParameterOrExistingIndex(5),
+                new AddedParameterOrExistingIndex(4),
+                new AddedParameterOrExistingIndex(3),
+                new AddedParameterOrExistingIndex(new AddedParameter(null, "byte", "bb", "34"), "byte"),
+                new AddedParameterOrExistingIndex(2),
+                new AddedParameterOrExistingIndex(1),
+                new AddedParameterOrExistingIndex(0)};
```

Comment:
1. do we also have trivia tests?
2. intead of random numbers, i think this would be a lot easier to specify the names of the parameters int the order we want.

Reply:
1. What do you mean? They all test trivia?
2. Filed https://github.com/dotnet/roslyn/issues/42734